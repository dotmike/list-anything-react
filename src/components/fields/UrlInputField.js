import React from "react";
import PropTypes from "prop-types";
import { Form } from "semantic-ui-react";

import InlineError from "../messages/InlineError";

const UrlInputField = ({
  value,
  editable,
  showFieldDescription,
  error,
  field,
  onChange
}) => {
  if (editable)
    return (
      <Form.Field error={!!error}>
        <Form.Input
          fluid
          type="url"
          label={field.description}
          value={value}
          onChange={onChange}
          name={field._id}
          placeholder="http://site.com/"
        />
        {error && <InlineError text={error} />}
      </Form.Field>
    );
  if (showFieldDescription)
    return (
      <Form.Field>
        <b>{field.description}:</b>{" "}
        <a href={value} target="_blank">
          {value}
        </a>
      </Form.Field>
    );
  return (
    <a href={value} target="_blank">
      {value}
    </a>
  );
};

UrlInputField.propTypes = {
  // ownProps
  value: PropTypes.string.isRequired,
  editable: PropTypes.bool,
  showFieldDescription: PropTypes.bool,
  error: PropTypes.string,
  field: PropTypes.shape({
    _id: PropTypes.string,
    description: PropTypes.string
  }),
  onChange: PropTypes.func
};

UrlInputField.defaultProps = {
  editable: false,
  showFieldDescription: true,
  error: "",
  field: {},
  onChange: () => {}
};

export default UrlInputField;
