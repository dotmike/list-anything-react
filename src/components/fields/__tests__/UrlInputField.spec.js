import React from "react";
import { shallow } from "enzyme";

import UrlInputField from "../UrlInputField";

const defaultProps = {
  value: "http://www.example.com",
  error: "",
  editable: true,
  showFieldDescription: false,
  field: { _id: "1", description: "Testando:" },
  onChange: () => {}
};

describe("UrlInputField", () => {
  describe("when `editable` is true", () => {
    const props = { ...defaultProps, editable: true };

    it("renders correctly", () => {
      const wrapper = shallow(<UrlInputField {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when `editable` is false", () => {
    let props;

    beforeEach(() => {
      props = { ...defaultProps, editable: false };
    });

    it("renders correctly when `showFieldDescription` is false", () => {
      props.showFieldDescription = false;

      const wrapper = shallow(<UrlInputField {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it("renders correctly when `showFieldDescription` is true", () => {
      props.showFieldDescription = true;

      const wrapper = shallow(<UrlInputField {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
