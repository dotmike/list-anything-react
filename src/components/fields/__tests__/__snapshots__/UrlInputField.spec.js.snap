// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlInputField when \`editable\` is false renders correctly when \`showFieldDescription\` is false 1`] = `
<a
  href="http://www.example.com"
  onClick={[Function]}
  target="_blank"
>
  http://www.example.com
</a>
`;

exports[`UrlInputField when \`editable\` is false renders correctly when \`showFieldDescription\` is true 1`] = `
<FormField>
  <b>
    Testando:
    :
  </b>
   
  <a
    href="http://www.example.com"
    onClick={[Function]}
    target="_blank"
  >
    http://www.example.com
  </a>
</FormField>
`;

exports[`UrlInputField when \`editable\` is true renders correctly 1`] = `
<FormField
  error={false}
>
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    label="Testando:"
    name="1"
    onChange={[Function]}
    placeholder="http://site.com/"
    type="url"
    value="http://www.example.com"
  />
</FormField>
`;
